config/dynamic.c

```diff
@@ -16,11 +16,467 @@
+}
+
+static void
+cleanup_dy(struct dynamic_config *dy_conf)
+{
+	int ret;
+
+	if (dy_conf->sock_fd >= 0) {
+		ret = close(dy_conf->sock_fd);
+		if (ret < 0)
+			perror("Failed to close the server socket");
+
+		dy_conf->sock_fd = -1;
+	}
+
+	if (dy_conf->lua_state != NULL) {
+		lua_close(dy_conf->lua_state);
+		dy_conf->lua_state = NULL;
+	}
+}
+
+static int
+dyn_cfg_proc(void *arg)
+{
+	int ret = 0;
+	int ip_reusable = 1;
+	struct sockaddr_in6 server_addr;
+	uint32_t lcore = rte_lcore_id();
+	struct dynamic_config *dy_conf = arg;
+
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: the Dynamic Config block is running at lcore = %u\n", lcore);
+
+	/* Init the server socket. */
+    	dy_conf->sock_fd = socket(AF_INET6, SOCK_STREAM, 0);
+	if (dy_conf->sock_fd < 0) {
+		exiting = true;
+		perror("Failed to initialize the server socket");
+		goto out;
+	}
+
+	if (setsockopt(dy_conf->sock_fd, SOL_SOCKET, SO_REUSEADDR,
+			(char *)&ip_reusable,sizeof(ip_reusable)) < 0) {
+		exiting = true;
+		perror("Failed to set setsockopt(SO_REUSEADDR)");
+		goto out;
+	}
+
+	/*
+	 * The Dynamic config component listens to a Unix socket
+	 * for request from the local host.
+	 */
+    	memset(&server_addr, 0, sizeof(server_addr));
+	server_addr.sin6_family = AF_INET6;
+    	server_addr.sin6_port = htons(dy_conf->sock_port);
+
+	/*
+	 * Setting the address to in6addr_any, this allows connections to
+	 * be established from any IPv4 or IPv6 client by default.
+	 */
+    	server_addr.sin6_addr = in6addr_any;
+
+    	ret = bind(dy_conf->sock_fd,
+		(struct sockaddr *)&server_addr, sizeof(server_addr));
+	if (ret < 0) {
+		exiting = true;
+		perror("Failed to initialize the server socket");
+		goto out;
+	}
+
+    	ret = listen(dy_conf->sock_fd, 10);
+	if (ret < 0) {
+		exiting = true;
+		perror("Failed to listen on the server socket");
+		goto out;
+	}
+
+	while (likely(!exiting)) {
+		fd_set fds;
+		struct timeval stv;
+ 
+		FD_ZERO(&fds);
+		FD_SET(dy_conf->sock_fd, &fds);
+ 
+		/*
+		 * 1000 usecs' timeout for the select() function.
+		 * This parameter can prevent the select() function
+		 * from blocking forever. So, the whole program can
+		 * exit when receiving a quitting signal.
+		 */
+		stv.tv_sec = 0;
+		stv.tv_usec = 1000;
+
+		ret = select(dy_conf->sock_fd + 1, &fds, NULL, NULL, &stv);
+		if (ret < 0 && errno != EINTR) {
+			perror("pselect()");
+			break;
+		} else if (unlikely(exiting)) {
+			break;
+		} else if (likely(ret == 0))
+			continue;
+
+		/*
+		 * The config component accepts only one connection at a time.
+		 */
+		if (FD_ISSET(dy_conf->sock_fd, &fds))
+			handle_client(dy_conf->sock_fd, dy_conf);
+	}
+
+out:
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: the Dynamic Config block at lcore = %u is exiting\n", lcore);
+
+	cleanup_dy(dy_conf);
+
+	return ret;
+}
+
+struct dynamic_config *
+alloc_dy_conf(void)
+{
+	return rte_calloc("dynamic_config",
+		1, sizeof(struct dynamic_config), 0);
+}
+
+static int
+dy_stage2(void *arg)
```

Comment:
Instead of keeping the same Lua state for all the life of Gatekeeper while running, it would be more robust to keep the Lua stat while there is a connection. This way, if a session of commands messes up the Lua state, all one needs to do is to establish another connection instead of restarting Gatekeeper.